src/components/datagrid/data_grid_cell.tsx

```diff
@@ -310,6 +321,14 @@ export class EuiDataGridCell extends Component<
     }
   };
 
+  closePopover = () => {
+    this.setState({ popoverIsOpen: false }, () =>
```

Comment:
This nested setState doesn't appear necessary for closing the popover

Reply:
I thought so as well originally, but it has the same problem (not always properly unmounted, leading to hanging popovers). I added a comment